vllm/model_executor/models/llama.py

```diff
@@ -363,9 +376,13 @@ def load_weights(self,
                     shard_size //= self.quant_config.pack_factor
                     offset //= self.quant_config.pack_factor
 
-                loaded_weight = loaded_weight[
-                    shard_size * tensor_model_parallel_rank:shard_size *
-                    (tensor_model_parallel_rank + 1)]
+                if weight_name != "q_proj":
```

Comment:
style nit: what about this?
```suggestion
                if weight_name in ["k_proj", "v_proj"]:
```